src/bootstrap/doc.rs

```diff
@@ -522,6 +522,76 @@ impl Step for Std {
     }
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct JsonStd {
+    pub stage: u32,
+    pub target: TargetSelection,
+}
+
+impl Step for JsonStd {
+    type Output = ();
+    const DEFAULT: bool = false;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        let builder = run.builder;
+        run.default_condition(builder.config.json_docs)
```

Comment:
you can remove `default_condition` altogether here, I think; I don't think this should ever be on by default.
```suggestion
        run.alias("rust-json-docs")
```